module/plugins/captcha/ReCaptcha.py

```diff
@@ -354,3 +356,45 @@ def _challenge_v2(self, key, secure_token=None):
             self.fail(_("Recaptcha max retries exceeded"))
 
         return result, challenge
+
+    # solve interactive captcha (javascript required), use when non-JS captcha fallback for v2 is not allowed
+    def _challenge_v2_js(self, key):
+        self.log_debug("challenge ReCaptcha v2 interactive")
+        captchaManager = self.pyload.captchaManager
+
+        try:
+            time_ref = ("%.2f" % time.time())[-6:].replace(".", "")
+            filename = "captcha_request_%s_%s.nofile" % (self.pyfile.plugin.__name__, time_ref)
+            # we don't have an actual image to decrypt -> pass the URL and the site-key for further use in plugins/hooks
+            img = {'url': self.pyfile.url, 'sitekey': self.retrieve_key(None)}
+            self.task = captchaManager.newTask(img, "interactive", filename , "interactive")
```

Comment:
pass `None` instead of filename as it is not used anyway. then you can remove line 366 and 367

Reply:
Thought the same, but I saw that e.g. the 9kw-Plugin uses the image-name to detect the hoster of the CaptchaTask.
"captcha_request_ShareOnlineBiz_0123456.jpg"
The plugin extracts "ShareOnlineBiz" to apply hoster-dependet settings of 9kw.

I don't know if any other plugins also rely on the informations "encoded" in the image-name

If there is a other way to get the hoster in the 9kw-plugin, then the image name can be set to None. Maybe this requires to add a "hoster"-attribute to the CaptchaTask-Class in CaptchaManager.py.
Passing this "filename" was just an easy workaround arount this ....